lib/ansible/modules/cloud/amazon/ec2_vpc_peer.py

```diff
@@ -419,20 +425,14 @@ def main():
         module.fail_json(msg="Can't authorize connection - " + str(e))
 
     if state == 'present':
-        if not vpc_id and not peer_vpc_id:
-            module.fail_json(msg='vpc_id and peer_vpc_id parameters are required for creating peering connection')
-
         (changed, results) = create_peer_connection(client, module)
         module.exit_json(changed=changed, peering_id=results)
     elif state == 'absent':
-        if not peering_id or (not vpc_id and not peer_vpc_id):
-            module.fail_json(msg='peering_id or vpc_id and peer_vpc_id parameters are required for removing peering connection')
+        if not peering_id and (not vpc_id or not peer_vpc_id):
+            module.fail_json(msg='state is absent but all of the following are missing: peering_id or [vpc_id, peer_vpc_id]')
```

Comment:
'all of the following are missing:' is probably a little difficult to understand - perhaps use 'one of the following is missing:'

It is difficult to word in a way that is clear though!

Reply:
Yeah -- don't really know what's the best way to go about this message, the old one seems better but doesn't really match new message with the other states

Reply:
Feel free to go back to the original message - we can keep the required_if for the other messages, we should be consistent where it makes sense but it's better to be readable!